lib/DiscoveryStreamFeed.jsm

```diff
@@ -32,12 +33,25 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
     this._prefCache = {};
   }
 
+  finalLayoutEndpoint(url, apiKey) {
+    if (url.includes("$apiKey") && !apiKey) {
+      throw new Error(`Layout Endpoint - An API key was specified but none configured: ${url}`);
```

Comment:
I just realized this is going to be caught by the outer try/catch... maybe make sure to log it or change the way the flow works so this will show up?

Reply:
Good catch, updated to log error in catch block